apiserver/facades/client/charms/client.go

```diff
@@ -822,8 +822,7 @@ func (a *API) listOneCharmResources(arg params.CharmURLAndOrigin) ([]params.Char
 	if err != nil {
 		return nil, apiservererrors.ServerError(err)
 	}
-
-	repo, err := a.getCharmRepository(corecharm.Source(curl.Schema))
+	repo, err := a.getCharmRepository(corecharm.CharmHub)
```

Comment:
Shouldn't this be

`repo, err := a.getCharmRepository(arg.Origin.Source)`

? 
Or else how does it work for cs charms?

Reply:
It doesn't.  Ln 813 returns an error if !charm.CharmHub.Matches(curl.Schema).  The charm repository's version of ListResources is a no-op.

The charm store request goes thru a very different code path. Which can be verified in QA and shown above.

However I have updated the code and made a long commit message about the details.